editor/editor_log.h

```diff
@@ -36,19 +36,92 @@
 #include "scene/gui/button.h"
 #include "scene/gui/control.h"
 #include "scene/gui/label.h"
+#include "scene/gui/line_edit.h"
 #include "scene/gui/panel_container.h"
 #include "scene/gui/rich_text_label.h"
 #include "scene/gui/texture_button.h"
 #include "scene/gui/texture_rect.h"
 
-class EditorLog : public VBoxContainer {
-	GDCLASS(EditorLog, VBoxContainer);
+class EditorLog : public HBoxContainer {
+	GDCLASS(EditorLog, HBoxContainer);
+
+public:
+	enum MessageType {
+		MSG_TYPE_STD,
+		MSG_TYPE_ERROR,
+		MSG_TYPE_WARNING,
+		MSG_TYPE_EDITOR,
+	};
+
+private:
+	struct LogMessage {
+		String text;
+		MessageType type;
+		int count = 1;
+
+		LogMessage() {}
+
+		LogMessage(const String p_text, MessageType p_type) :
+				text(p_text),
+				type(p_type) {
+		}
+	};
+
+	// Encapsulates all data and functionality regarding filters.
+	struct LogFilter {
+	private:
+		// Force usage of set method since it has functionality built-in.
+		int message_count = 0;
+
+	public:
+		MessageType type;
+		Button *toggle_button = nullptr;
+		bool active = true;
+
+		void initialise_button(const String &p_tooltip, Callable p_toggled_callback) {
```

Comment:
Sorry Commonwealth, the US won the IT wars.
```suggestion
		void initialize_button(const String &p_tooltip, Callable p_toggled_callback) {
```

Reply:
Is there a specific reason to implement this in the header instead of the `.cpp`?